src/util.cpp

```diff
@@ -187,6 +187,20 @@ bool DirIsWritable(const fs::path& directory)
     return true;
 }
 
+bool GetRelativePath(fs::path& result, const fs::path& path, const fs::path& base)
+{
+    if (!path.is_absolute() || !base.is_absolute()) return false;
+    fs::path relative, parent = path;
+    boost::system::error_code ec;
+    while (!parent.empty() && !fs::equivalent(parent, base, ec)) {
+        relative = parent.filename() / relative;
+        parent = parent.parent_path();
```

Comment:
On Windows, `std::filesystem::path("C:\\").parent_path() == std::filesystem::path("C:\\")`. It would be an infinity loop. (Only test std::filesystem in MSVC, doesn't test boost::filesystem)

Reply:
Added `parent != parent.root_path()` condition to the loop.

Reply:
> On Windows, std::filesystem::path("C:\\").parent_path() == std::filesystem::path("C:\\")

Workaround seems ok, but is this is a bug? Both boost filesystem and std::filesystem are supposed to return empty if the path contains a single element:

https://en.cppreference.com/w/cpp/experimental/fs/path/parent_path
https://www.boost.org/doc/libs/1_68_0/libs/filesystem/doc/reference.html#path-parent_path

Reply:
https://en.cppreference.com/w/cpp/filesystem/path/parent_path
The C++17 standard one is a little different from the experimental one.